### ISSUE TITLE
Rule request: disallow nested landmark elements of the same name

### ISSUE THREAD

**Issue by MelSumner** [ember-template-lint/ember-template-lint#1274]
A new rule should be added to disallow landmark elements to be nested within landmark elements of the same name.

## List of elements & their corresponding roles
- header (banner)
- main (main)
- aside (complementary)
- form (form, search)
- main (main)
- nav (navigation)
- footer (contentinfo)

### Allowed

```html
<main></main>
```

### Disallowed

```html
<main>
  <main></main>
</main>
```

```html
<main>
  <div role="main"></div>
</main>
```

**Comment by rajasegar** [comment]
I am picking this one. @MelSumner Please assign it to me.
